erpnext/stock/doctype/delivery_trip/delivery_trip.js

```diff
@@ -66,6 +66,60 @@ frappe.ui.form.on('Delivery Trip', {
 		}
 	},
 
+	validate: function (frm) {
+		if (frm.is_new() && frm.doc.delivery_stops) {
+			// frappe.confirm doesn't wait for callback to validate, so creating
+			// Promise to make sure execution is paused until user input
+			return new Promise((resolve, reject) => {
+				frappe.call({
+					method: "erpnext.stock.doctype.delivery_trip.delivery_trip.validate_unique_delivery_notes",
+					args: {
+						delivery_stops: frm.doc.delivery_stops
+					},
+					callback: (r) => {
+						if (!$.isEmptyObject(r.message)) {
+							// create an unordered list of existing note <-> trip mappings
+							let existing_trips = "<ul>";
+							for (let trip in r.message) {
+								existing_trips += `<li><strong>${trip}</strong> exists in <strong>${r.message[trip]}</strong></li>`
+							}
+							existing_trips += "</ul>"
+
+							let confirm_message = `The following deliveries already exist in recent
+								trip(s): ${existing_trips} Do you still want to continue?`;
+
+							frappe.confirm(
+								__(confirm_message),
+								() => { resolve(); },  // If "Yes" is selected
+								() => { frappe.set_route("List", frm.doc.doctype); }  // If "No" is selected
+							);
+						} else { resolve(); }
+					},
+					error: (r) => reject(r.message)
+				})
+			})
+		}
+	},
+
+	driver: function (frm) {
+		if (frm.doc.driver) {
+			frappe.db.get_value("Delivery Trip", {
+				docstatus: ["<", 2],
+				driver: frm.doc.driver,
+				departure_time: [">", frappe.datetime.nowdate()]
+			}, "name", (r) => {
+				if (r) {
+					let confirm_message = `${frm.doc.driver_name} has already been assigned a Delivery Trip
```

Comment:
translations

Reply:
Already done on line 115.

Reply:
Moved translations to string generation.